server/src/main/java/org/elasticsearch/index/store/FsDirectoryFactory.java

```diff
@@ -154,22 +156,38 @@ protected boolean useDirectIO(String name, IOContext context, OptionalLong fileL
 
         @Override
         public IndexInput openInput(String name, IOContext context) throws IOException {
+            Throwable directIOException = null;
             if (directIODelegate != null && context.hints().contains(DirectIOHint.INSTANCE)) {
                 ensureOpen();
                 ensureCanRead(name);
-                Log.debug("Opening {} with direct IO", name);
-                return directIODelegate.openInput(name, context);
-            } else if (useDelegate(name, context)) {
-                // we need to do these checks on the outer directory since the inner doesn't know about pending deletes
-                ensureOpen();
-                ensureCanRead(name);
-                // we only use the mmap to open inputs. Everything else is managed by the NIOFSDirectory otherwise
-                // we might run into trouble with files that are pendingDelete in one directory but still
-                // listed in listAll() from the other. We on the other hand don't want to list files from both dirs
-                // and intersect for perf reasons.
-                return delegate.openInput(name, context);
-            } else {
-                return super.openInput(name, context);
+                try {
+                    Log.debug("Opening {} with direct IO", name);
```

Comment:
How does this affect the DirectIOIT test ?   Maybe we wanna make the test assertions conditional rather than skip? That way we can assert the fallback case, no?

Reply:
Currently the direct IO test is unaffected, as that doesn't run the test at all if this would have thrown. We could include the negative test - that the exception is logged if direct IO is not supported?

Reply:
++ let's add that log message. Since we want to assert it in some conditions.  So, for test, instead of assuming directIO and only running if it is supported, we change the test to always run and assert the expected log message, right?

Reply:
Added